apps/prairielearn/src/admin_queries/course_instance_active_enrollments.sql

```diff
@@ -1,124 +1,135 @@
+      institutions AS i
+      JOIN pl_courses AS c ON (c.institution_id = i.id)
+      JOIN course_instances AS ci ON (ci.course_id = c.id)
+      JOIN assessments AS a ON (a.course_instance_id = ci.id)
       JOIN assessment_instances AS ai ON (ai.assessment_id = a.id)
       JOIN instance_questions AS iq ON (iq.assessment_instance_id = ai.id)
       JOIN users AS u ON (u.user_id = iq.authn_user_id)
-      JOIN pl_courses AS c ON (c.id = ci.course_id)
-      JOIN institutions AS i ON (i.id = c.institution_id)
+    WHERE
+      i.short_name = $institution_short_name
+      AND ai.modified_at BETWEEN $start_date AND $end_date
+      AND ai.include_in_statistics
+      AND iq.modified_at BETWEEN $start_date AND $end_date
     GROUP BY
       ci.id,
       u.user_id,
       i.id,
       u.institution_id
   ),
-  course_instance_user_selection AS (
-    SELECT
-      *
-    FROM
-      course_instance_user_data
-    WHERE
-      NOT users_is_instructor_in_course_instance (user_id, id)
-      AND instance_question_count >= $minimum_instance_question_count
-  ),
   course_instance_data AS (
     SELECT
-      id,
+      course_instance_id,
       count(*) AS total_students,
       count(*) FILTER (
         WHERE
           NOT is_institution_user
       ) AS outside_students,
-      count(*) FILTER (
-        WHERE
-          instance_question_within_dates_count >= $minimum_instance_question_count
-      ) AS within_dates_students,
-      sum(instance_question_count) AS instance_question_count,
-      avg(
-        instance_question_within_dates_count::double precision / instance_question_count::double precision * 100
-      ) AS activity_within_dates_perc,
-      percentile_disc(0.5) WITHIN GROUP (
-        ORDER BY
-          start_date
-      ) AS start_date,
-      percentile_disc(0.5) WITHIN GROUP (
-        ORDER BY
-          end_date
-      ) AS end_date
+      sum(instance_question_count) AS instance_question_count
+    FROM
+      course_instance_user_data
+    WHERE
+      instance_question_count >= $minimum_instance_question_count
+    GROUP BY
+      course_instance_id
+  ),
+  workspace_data AS (
+    SELECT
+      ci.id AS course_instance_id,
+      COUNT(*) AS workspace_count,
+      sum(
+        extract(
+          epoch
+          from
+            w.launching_duration + w.running_duration
+        ) / 3600.0
+      ) AS workspace_hours
+    FROM
+      institutions AS i
+      JOIN pl_courses AS c ON (c.institution_id = i.id)
+      JOIN course_instances AS ci ON (ci.course_id = c.id)
+      JOIN assessments AS a ON (a.course_instance_id = ci.id)
+      JOIN assessment_instances AS ai ON (ai.assessment_id = a.id)
+      JOIN instance_questions AS iq ON (iq.assessment_instance_id = ai.id)
+      JOIN variants AS v ON (v.instance_question_id = iq.id)
+      JOIN workspaces AS w ON (w.id = v.workspace_id)
+    WHERE
+      i.short_name = $institution_short_name
+      AND w.created_at BETWEEN $start_date AND $end_date
```

Comment:
Both the workspace and external grading queries use a more exhaustive method of finding all workspaces and grading jobs within the date range. These queries don't use `ai.modified_at` or `iq.modified_at` so they really will find all compute activity in the date range, no matter what.